### DISCUSSION TITLE
How to configure idle_timeout when using ./run.sh --jitconfig?

### DISCUSSION THREAD

**Discussion by geekflyer** [community/community#60513] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi,

I'm trying to spawn self hosted runners which use --jitconfig and are being started via `./run.sh --jitconfig`. 
In my setup a JIT runner is usually spawned for each individual job. Sometimes jobs get cancelled before a JIT runner picks up its designated job and then the JIT runner stays idle for a long time.

So I'd like to set the idle timeout to some very short value (10 seconds or whatever).

How can I do that?

@Envigit mentioned the idle_timeout config in some config file (what's the name of that file anyways) here https://github.com/orgs/community/discussions/50926  but I'm not quite sure how can I use that with jitconfig since jit config can only be started via `./run.sh` and there doesn't appear to be a config file.

**Comment by ChristopherHX** [comment, marked as answer]
I don't think that `idle_timeout` exists in actions/runner, search for `idle_timeout` in actions/runner = 0 results.

Also actions/runner doesn't seem to support graceful exit on idle.

_Unsupported content, behavior is not document use at your own risk_
I remember
- that active runners cannot be removed
- call https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#delete-a-self-hosted-runner-from-a-repository with the runner id after 10s **unconditionally** (I expect that you get a http error if it failed, just ignore that during test today is response bad request, use a higher timeout if you need to start external container / VM's)
- An idle runner will exit, because it has been removed from Actions Service
- Maybe just do this where you generate the jitconfig, the runner id is a field in the jitconfig rest response

I suggest to make a feature request discussion for an officially supported way.

**Reply by geekflyer** [reply]
hmm, seems like you're right. There's not even a single result of even `idle` or `idle_timeout` in the codebase. The answer from @enviGit in that other channel sounded quite "official" but now in hindsight I realize this might be just some confident sounding AI generated response lol.
